OrbitGl/BatcherTest.cpp

```diff
@@ -28,46 +28,50 @@ class MockBatcher : public Batcher {
   // buffers. Only a single color per element will be appended
   // (start point for line, first vertex for triangle and box)
   void Draw(bool picking = false) const override {
-    if (picking) {
-      for (auto it = line_buffer_.picking_colors_.begin();
-           it != line_buffer_.picking_colors_.end();) {
-        drawn_line_colors_.push_back(*it);
-        ++it;
-        ++it;
-      }
-      for (auto it = triangle_buffer_.picking_colors_.begin();
-           it != triangle_buffer_.picking_colors_.end();) {
-        drawn_triangle_colors_.push_back(*it);
-        ++it;
-        ++it;
-        ++it;
-      }
-      for (auto it = box_buffer_.picking_colors_.begin();
-           it != box_buffer_.picking_colors_.end();) {
-        drawn_box_colors_.push_back(*it);
-        ++it;
-        ++it;
-        ++it;
-        ++it;
-      }
-    } else {
-      for (auto it = line_buffer_.colors_.begin(); it != line_buffer_.colors_.end();) {
-        drawn_line_colors_.push_back(*it);
-        ++it;
-        ++it;
-      }
-      for (auto it = triangle_buffer_.colors_.begin(); it != triangle_buffer_.colors_.end();) {
-        drawn_triangle_colors_.push_back(*it);
-        ++it;
-        ++it;
-        ++it;
-      }
-      for (auto it = box_buffer_.colors_.begin(); it != box_buffer_.colors_.end();) {
-        drawn_box_colors_.push_back(*it);
-        ++it;
-        ++it;
-        ++it;
-        ++it;
+    for (auto& [unused_layer, buffer] : buffers_by_layer_) {
+      if (picking) {
+        for (auto it = buffer.line_buffer.picking_colors_.begin();
+             it != buffer.line_buffer.picking_colors_.end();) {
+          drawn_line_colors_.push_back(*it);
+          ++it;
+          ++it;
+        }
+        for (auto it = buffer.triangle_buffer.picking_colors_.begin();
+             it != buffer.triangle_buffer.picking_colors_.end();) {
+          drawn_triangle_colors_.push_back(*it);
+          ++it;
+          ++it;
+          ++it;
```

Comment:
nit. the code was probably already there, but can't we do "it+=2" and "it+=3" here?

Reply:
it is BlockIterator, and we don't have implemented the "+" operator there, so we can't make it+=n